### PR TITLE
docs: rewrite source types document

### PR DESCRIPTION
**Pull request by TC-MO**
improve structure
enhanced clarity & simplified language
added additional context
improved styling & formatting

### CODE REVIEW

File: sources/platform/actors/development/deployment/index.md [hunk 1]
```diff
@@ -5,35 +5,67 @@ description: Learn how to deploy your Actor to the Apify platform and build them
...
+
+To deploy your Actor using Apify CLI:
+
+1. Log in to your Apify account:
+
+    ```bash
+    apify login
+    ```
+
+1. Navigate to the directory of your Actor on your local machine.
```
Comment (drobnikj) [inline sources/platform/actors/development/deployment/index.md:26]:
```suggestion
2. Navigate to the directory of your Actor on your local machine.
```
Numbers are not correct from here.

File: sources/platform/actors/development/deployment/index.md [hunk 2]
```diff
@@ -5,35 +5,67 @@ description: Learn how to deploy your Actor to the Apify platform and build them
...
+
+1. Deploy your Actor by running:
```
Comment (drobnikj) [inline sources/platform/actors/development/deployment/index.md:28]:
```suggestion
3. Deploy your Actor by running:
```

File: sources/platform/actors/development/deployment/source_types.md [hunk 3]
```diff
@@ -5,69 +5,92 @@ slug: /actors/development/deployment/source-types
...
-[//]: # (TODO: It's pretty outdated, we should probably update the Actor too)
-To help you get started quickly, you can use the [apify/quick-start](https://apify.com/apify/quick-start) Actor which contains all the boilerplate necessary when creating a new Actor hosted on Git. The source code is available on [GitHub](https://github.com/apify/actor-quick-start).
+#### How to configure deployment keys
 
-To specify a Git branch or tag to check out, add a URL fragment to the URL. For example, to check out the **develop** branch, specify a URL such as `https://github.com/jancurn/some-actor.git#develop`.
+To configure the deployment key for your Actor's source code you need to:
 
-Optionally, the second part of the fragment in the Git URL (separated by a colon) specifies the directory from which the Actor will be built (and where the `.actor`) folder is located. For example, `https://github.com/jancurn/some-actor.git#develop:some/dir` will check out the **develop** branch and set **some/dir** as the root directory of the Actor.
+1. In your Actor's settings, find the **Git URL** input field
+1. Click the **deployment key** link below the input field
```
Comment (drobnikj) [inline sources/platform/actors/development/deployment/source_types.md:58]:
```suggestion
2. Click the **deployment key** link below the input field
```

File: sources/platform/actors/development/deployment/source_types.md [hunk 4]
```diff
@@ -5,69 +5,92 @@ slug: /actors/development/deployment/source-types
...
+1. Follow the instruction to add the key to your Git hosting service.
```
Comment (drobnikj) [inline sources/platform/actors/development/deployment/source_types.md:59]:
```suggestion
3. Follow the instruction to add the key to your Git hosting service.
```

**Review by drobnikj — APPROVED** [review]
Looks good, just please fix the list numbers.

File: sources/platform/actors/development/deployment/source_types.md (hunk 4 above)
Comment (TC-MO) [reply sources/platform/actors/development/deployment/source_types.md:59]: Docusaurus would change those into proper numbers, but I guess we can have it this way also. Will fix, thank you!

**Pull request merged**